Cloud Docs (#776)
grouparoo/www.grouparoo.com: pages/docs/cloud/repository-integration.mdx

---
title: "Grouparoo Cloud Repository Integration"
date: "2021-11-10"
pullQuote: "Simple integration steps for Grouparoo Cloud"
---

Your Project in [Grouparoo Cloud](https://cloud.grouparoo.com/) has an Integrations Page.

<Image
  src="docs/cloud/project-landing-page-click-integrations.png"
  alt="Cloud App Fresh Project Landing Page Click Integrations"
  width={1544}
  height={469}
/>

> If has instructions on what you need from Cloud to integrate with your GitHub repository.

Reviewer: ```suggestion
This page has instructions on what you need from Cloud to integrate with your GitHub repository.
```

> ---

Reviewer 2: Please add this page to the sidebar under `Cloud` so people can get to it more easily

title: "Grouparoo Cloud Repository Integration"
date: "2021-11-10"
pullQuote: "Simple integration steps for Grouparoo Cloud"
---

If you are here you should already have an API Key and Project ID.

Grouparoo Cloud requires three things to function: A github action yaml file and two secrets to identify and authorize the code config repository.

> ### Github Repository Secrets

Reviewer 2: ```suggestion
### GitHub Repository Secrets
```

Grouparoo Cloud requires two secrets to be available in your repository: `GROUPAROO_CLOUD_TOKEN` and `GROUPAROO_CLOUD_PROJECT_ID`.

> If these values have already been provided for you, skip ahead. Otherwise you can get them from [Grouparoo Cloud](https://cloud.grouparoo.com/) in your Project's Integrations page.

Reviewer 2: "skip ahead" is not quite accurate- even though you have the secrets you still need to set them on the repository

> In your Github repository go to: Settings -> Secrets -> New repository secret.

Reviewer 2: Let's show a screenshot here to make sure people know this is on the GitHub side. When trying to follow this with Andy, he got a bit confused on where exactly this was referring to

> Add `GROUPAROO_CLOUD_TOKEN` and `GROUPAROO_CLOUD_PROJECT_ID` with the provided values.

Reviewer 2: Maybe we can display a table of sorts that has a short description on what each secret is

Author: Do we have a particular pattern for this?

Reviewer 2: Not really, but I think now that we have the Integrations page and mentioning that you can get them from there makes more sense. So maybe it's not needed anymore.

Author: I could just say `Copy the values from your Project's integrations page`?

<sub>The Project ID should look similar to `prj_some-set-of-values`.</sub>

> ### Github Action

Reviewer 2: ```suggestion
### GitHub Action
```

> We recommend creating a new branch for this since our system can check PR builds for validity, saving you some potential headache.

Reviewer 2: The addition of the github action doesn't necessarily need to be on a new branch- at this point in the process there's nothing to validate yet. I think this can be an opportunity to explain a bit about what this action does and why you may want to install it, highlighting that Grouparoo Cloud can auto validate the config changes through PRs as well as deploy them once merged into main.

Author: The thought for these docs is you already have an existing code config repo. I've tried to clarify that at the beginning.

Create a file in your repository: `.github/workflows/grouparoo-cloud.yml`
And place the below code into it.

```yml
name: grouparoo-cloud

on:
  push:
    branches:
      - main
  pull_request:
    branches:
      - main

jobs:
  ci:
    if: ${{ github.event_name == 'pull_request' }}
    runs-on: ubuntu-latest
    steps:

>       - uses: actions/checkout@master

Reviewer 2: ```suggestion
      - uses: actions/checkout@v2
```

Author: Should I do this in both `ci` and `deploy`?

Reviewer 2: yep!

        with:
          ref: ${{ github.head_ref }}
      - uses: grouparoo/cloud-action@main
        with:
          project_id: ${{ secrets.GROUPAROO_CLOUD_PROJECT_ID }}
          token: ${{ secrets.GROUPAROO_CLOUD_TOKEN }}

>           api_url: "https://staging-cloud.grouparoo.com"

Reviewer 2: ```suggestion
```

Defaults to `cloud.grouparoo.com`

  deploy:
    if: ${{ github.event_name == 'push' }}
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@master
        with:
          ref: ${{ github.head_ref }}
      - uses: grouparoo/cloud-action@main
        with:
          type: deploy
          project_id: ${{ secrets.GROUPAROO_CLOUD_PROJECT_ID }}
          token: ${{ secrets.GROUPAROO_CLOUD_TOKEN }}

>           api_url: "https://staging-cloud.grouparoo.com"

Reviewer 2: ```suggestion
```

```

> Then create a pull request against your main branch. It may take a few minutes but our CI system should kick in and confirm that your Code Config is valid and hostable. If CI runs green, merge and our system will deploy it into Grouparoo Cloud. Once it's available your Project page in Grouparoo Cloud will provide a link!

Reviewer 2: I think this step should go after adding your grouparoo config. There could probably be another section in this page _after_ "GitHub Action" that tells you to initialize the project and make a PR for it, showing that validation will run on the PR.

Author: I added a sentence at the top linking to our Config UI docs. Ideally they'll have a code config repo already setup, and if not, those docs should walk them through it no?

Reviewer 2: Yep, this makes more sense! 

`Once it's available your Project page in Grouparoo Cloud will provide a link!`
Technically they should already have that link as soon as they have access to grouparoo cloud.

Reviewer: ```suggestion
Then create a pull request against your main branch. It may take a few minutes but our CI system will kick in and confirm that your Code Config is valid and host-able. If CI runs green, merge and our system will deploy it into Grouparoo Cloud. Once it's available your Project page in Grouparoo Cloud will provide a link!
```

---
title: "Grouparoo Cloud Repository Integration"
date: "2021-11-10"
pullQuote: "Simple integration steps for Grouparoo Cloud"
---

> Grouparoo Cloud requires three things to function: A github action yaml file and two secrets to identify and authorize the code config repository.

Reviewer: ```suggestion
Grouparoo Cloud requires three things to function: A github action `.yaml` file and two secrets to identify and authorize the code config repository.
```

It is assumed you have an existing repository on github that was setup using our [Config UI](/docs/config).

### GitHub Repository Secrets

Grouparoo Cloud requires two secrets to be available in your repository: `GROUPAROO_CLOUD_TOKEN` and `GROUPAROO_CLOUD_PROJECT_ID`.

If these values have not already been provided for you, you can get them from [Grouparoo Cloud](https://cloud.grouparoo.com/) in your Project's Integrations page.

In your GitHub repository go to: Settings -> Secrets -> New repository secret.

> <Image

Reviewer 2: Nice!

  src="docs/cloud/action-secrets.png"
  alt="GitHub Repository Secrets"
  width={1127}
  height={650}
/>

Add `GROUPAROO_CLOUD_TOKEN` and `GROUPAROO_CLOUD_PROJECT_ID` with the provided values.

> <sub>The Project ID should look similar to `prj_some-set-of-values`.</sub>

Reviewer: Why is this sentence `<sub>`? Seems like the backticks also don't work here.

Author: To function more like a footnote. If the backticks are showing through I can remove them.
